sale_order_general_discount_triple/models/sale_order_line.py

```diff
@@ -52,3 +52,31 @@ def create(self, vals):
         if general_discount:
             vals[general_discount] = sale_order.general_discount
         return super().create(vals)
+
+    @api.onchange("product_id", "product_uom_qty", "product_uom")
+    def _onchange_product_id_discount(self):
```

Comment:
@RabbitJon-S73 I would avoid onchanges by changing fields properties to compute, stored=True, readonly=False